### PR TITLE
Refactor Marketplace and Theme Components, Improve State and URL Handling

### PR DESCRIPTION
**Pull request by TheRealMG**
### Enhancements:
- **theme-card**: Changed cursor behavior on hover by adding the `cursor-pointer` class.
- **useMarketplace**: Improved maintainability by abstracting 100+ lines of logic from MarketplacePage into a reusable hook, allowing better separation of concerns between UI and business logic.

### Refactors:
- **theme-card**: Extract inline `onMouseDown` handler to improve readability. Moved the function to a separate handler named `handleMouseDown`.
- **marketplace**: Replace `useRef` with `useState` for `selectedTags` to improve state management and reactivity.
- **marketplace**: Improve side effects for pagination and theme loading by moving `currentPage` clamping logic to a dedicated function and optimizing `useEffect` for `loadedThemes`.
- **marketplace**: Extract MarketplacePage logic into custom hook (useMarketplace). Moved state management, filtering, sorting, and pagination logic from MarketplacePage to a custom hook, simplifying component structure and improving maintainability.
- **useMarketplace**: Streamline state management and URL updates. Replaced manual URL construction with updateSearchParams to simplify query parameter updates. Removed redundant state tracking (loadedThemes) and optimized pagination, sorting, and filtering logic.

### Fixes:
- **marketplace**: Synchronize tag toggling with URL search params. Resolved issue where search params were out of sync by using the functional form of `setSelectedTags`.

### CODE REVIEW

**Comment by TheRealMG** [comment]
Should be good to merge this now, I think I've changed all I wanted to

File: src/components/theme-card.tsx [hunk 1]
```diff
@@ -22,19 +22,21 @@ export default function ThemeCard({
...
-				//   please for the love of god, dont change this to a Link component. Please.
-				//
-				// If you had the tentation to change this to a Link component and saw this comment, please
-				//   increase this number: 1
-				if (e.button !== 0 && e.button !== 1) return;
-				if (e.target instanceof HTMLAnchorElement) return;
-				window.open(`/mods/${theme.id}`, e.button === 1 ? "_blank" : "_self");
-			}}
+			className="h-full cursor-pointer select-none flex-col justify-between rounded-xl border-2 border-[transparent] bg-surface transition-all duration-200 hover:border-[rgba(0,0,0,.5)] hover:shadow-lg dark:bg-[#121212] dark:hover:border-[#333]"
+			onMouseDown={handleMouseDown}
```
Comment (mauro-balades) [inline src/components/theme-card.tsx:34]: Please move the comment to inside the function

**Review by mauro-balades — CHANGES_REQUESTED** [review]

File: src/components/theme-card.tsx (hunk 1 above)
Comment (NOCanoa) [reply src/components/theme-card.tsx:34]: @mauro-balades  here? 
image

Comment (mauro-balades) [reply src/components/theme-card.tsx:34]: on top

Comment (NOCanoa) [reply src/components/theme-card.tsx:34]: in the top of const or top top of the function itself
image

Comment (NOCanoa) [reply src/components/theme-card.tsx:34]: good now? 
image

**Pull request merged**